### ISSUE TITLE
Not possible to alter send and recieve timeouts in \Kafka\Socket w/o forking code base

### ISSUE THREAD

**Issue by Crim** [nmred/kafka-php#27]
So its not possible to configure the send and receive timeouts on the socket select call w/o forking the code base and modifying the defaults defined in that class.

**Comment by Crim** [comment]
Submitted PR to make these options configurable from the client class, and accessible via Producer and Consumer classes 

https://github.com/nmred/kafka-php/pull/28

**Comment by Crim** [comment]
Additionally - On systems that have heavy write loads to Kafka with larger payloads, its possible to overrun the socket send buffer faster than it can get pushed out over the wire.  When this happens, the select() call in \Kafka\Socket::write() will return 0 showing that the stream is not available for writing (because the send buffer is full) and the code will throw a Kafka\Socket exception.  

In this case there is no graceful way I've found to handle the exception, and retry the send.  Building ontop of PR #28 is this guy: https://github.com/Crim/kafka-php/compare/master...Crim:selectRetry?expand=1

In my local testing I've found that typically after 1 additional retry, the write completes successfully.  Clearly there's a balance here between the number of retries and the send timeout options that will determine how long this method will block the processing thread of PHP.

Is there a more graceful way to solve this problem?  I played around with adjust send buffer sizes on the stream/socket, but that didn't really seem to help.
